sdx_controller/controllers/connection_controller.py

```diff
@@ -31,7 +32,30 @@ def delete_connection(connection_id):
 
     :rtype: None
     """
-    return "do some magic!"
+    logger.info(
+        f"Handling delete (connecton id: {connection_id}) "
+        f"with te_manager: {current_app.te_manager}"
+    )
+
+    # # Looking up by UUID do not seem work yet.  Will address in
+    # # https://github.com/atlanticwave-sdx/sdx-controller/issues/252.
+    #
+    # value = db_instance.read_from_db(f"{connection_id}")
+    # print(f"value: {value}")
+    # if not value:
+    #     return "Not found", 404
+
+    try:
+        # TODO: pce's unreserve_vlan() method silently returns even if the
+        # connection_id is not found.  This should in fact be an error.
+        #
+        # https://github.com/atlanticwave-sdx/pce/issues/180
+        current_app.te_manager.unreserve_vlan(connection_id)
```

Comment:
The failure handler is in `handlers/connection_handler.py`, I guess it will not be able to call this to unreserve the VLAN? We'll need some other way to get the te_manager. Am I right?

Reply:
Yeah, I also like the idea of making PCE a service. That way it saves us from the effort to pass PCE instances around.

Reply:
There are trade-offs to making PCE a service, so we should be careful there. 

We will have to keep an extra service running, which will add latency.  Making PCE a service also will have maintenance and debugging costs, even if the development effort is similar to that of keeping it in the current form.